source/fe/mapping_q1.cc

```diff
@@ -1779,86 +1875,62 @@ transform_real_to_unit_cell_internal
-          for (unsigned int i=0; i<dim; ++i)
-            p_unit_trial[i] -= step_length * delta[i];
-
-          // shape values and derivatives
-          // at new p_unit point
-          compute_shapes(std::vector<Point<dim> > (1, p_unit_trial), mdata);
-
-          // f(x)
-          Point<spacedim> p_real_trial = transform_unit_to_real_cell_internal(mdata);
-          const Tensor<1,spacedim> f_trial = p_real_trial-p;
+      /*
+      Return x_unit_n if the norm of the correction is less than eps,
+      i.e. if further correction of x_unit_n is smaller than the tolerance value. 
+      */
 
 #ifdef DEBUG_TRANSFORM_REAL_TO_UNIT_CELL
-          std::cout << "     step_length=" << step_length << std::endl
-                    << "       ||f ||   =" << f.norm() << std::endl
-                    << "       ||f*||   =" << f_trial.norm() << std::endl
-                    << "       ||f*||_A =" << (df_inverse * f_trial).norm() << std::endl;
-#endif
-
-          // see if we are making progress with the current step length
-          // and if not, reduce it by a factor of two and try again
-          //
-          // strictly speaking, we should probably use the same norm as we use
-          // for the outer algorithm. in practice, line search is just a
-          // crutch to find a "reasonable" step length, and so using the l2
-          // norm is probably just fine
-          if (f_trial.norm() < f.norm())
-            {
-              p_real = p_real_trial;
-              p_unit = p_unit_trial;
-              f = f_trial;
-              break;
-            }
-          else if (step_length > 0.05)
-            step_length /= 2;
-          else
-            AssertThrow (false,
-                         (typename Mapping<dim,spacedim>::ExcTransformationFailed()));
-        }
-      while (true);
-
-      ++newton_iteration;
-      if (newton_iteration > newton_iteration_limit)
-        AssertThrow (false,
-                     (typename Mapping<dim,spacedim>::ExcTransformationFailed()));
-      last_f_weighted_norm = (df_inverse * f).norm();
+  std::cout << "Iteration = " << newton_iteration << std::endl
+            << "\tNorm: ||p_real_n - p|| = " << (p_real_n-p).norm() << std::endl 
+            << "\tNorm of the correction: ||delta_x_unit|| = ||x_" << newton_iteration + 1 << " - x_" << newton_iteration << "|| = " << delta_x_unit.norm() << std::endl
+	    << "\t||delta|| / eps = " << delta_x_unit.norm()/eps << ", where eps = " << eps << std::endl;     
+#endif 
+      
+      if (delta_x_unit.norm() < eps) 
+      return x_unit_n; 
+      
+/*      
+    Otherwise, correct x_unit_n with delta_x_unit and do another iteration.
+	  The spacedim-th component of transformed point is simply ignored 
+	  in the codimension one case. When this component is not zero, then we 
+	  are projecting the point to the surface or curve identified by the cell.*/
+
+	  for (unsigned int i=0; i<dim; ++i)
+		x_unit_n[i] += delta_x_unit[i];
```

Comment:
Just use `x_unit_n += delta_x_unit;` for the whole tensor. There's no need to do it for components individually.